### PR TITLE
feat(bootstrap): add kms option to cdk bootstrap

### LINKED ISSUE #2203: Add kms encryption key as argument for cdk bootstrap
The `cdk bootstrap` command creates the following resource in CloudFormation:

```
  StagingBucket:
    Type: AWS::S3::Bucket
    Properties:
      AccessControl: Private
      BucketEncryption:
        ServerSideEncryptionConfiguration:
          - ServerSideEncryptionByDefault:
              SSEAlgorithm: aws:kms
```

I have not seen any option to specify the KMS key to be used. Is there a way to do so?

Cheers!

### PR DESCRIPTION
**Pull request by hoegertn**
fixes #2203 and supersedes #2245 
I had to create a new PR as merging was not easy and I cannot change the other PR

Thanks to @abelmokadem for the work.

### CODE REVIEW

File: packages/aws-cdk/bin/cdk.ts [hunk 1]
```diff
@@ -51,13 +51,14 @@ async function parseCommandLineArguments() {
     .command([ 'synthesize [STACKS..]', 'synth [STACKS..]' ], 'Synthesizes and prints the CloudFormation template for this stack', yargs => yargs
       .option('exclusively', { type: 'boolean', alias: 'e', desc: 'only deploy requested stacks, don\'t include dependencies' }))
     .command('bootstrap [ENVIRONMENTS..]', 'Deploys the CDK toolkit stack into an AWS environment', yargs => yargs
-      .option('toolkit-bucket-name', { type: 'string', alias: 'b', desc: 'The name of the CDK toolkit bucket', default: undefined }))
+      .option('toolkit-bucket-name', { type: 'string', alias: 'b', desc: 'The name of the CDK toolkit bucket', default: undefined })
+      .option('toolkit-bucket-kms-key-id', { type: 'string', desc: 'AWS KMS master key ID used for the SSE-KMS encryption', default: undefined }))
```
Comment (eladb) [inline packages/aws-cdk/bin/cdk.ts]: rename to `bootstrap-kms-key-id`

Comment (eladb) [inline packages/aws-cdk/bin/cdk.ts]: add `--bootstrap-bucket-name` as an alias to `--toolkit-bucket-name`

Comment (hoegertn) [reply packages/aws-cdk/bin/cdk.ts]: How can I add a second alias to `--toolkit-bucket-name`?

Comment (eladb) [reply packages/aws-cdk/bin/cdk.ts]: I think there should be an .alias() method

File: packages/aws-cdk/bin/cdk.ts [hunk 2]
```diff
@@ -51,7 +51,7 @@ async function parseCommandLineArguments() {
     .command([ 'synthesize [STACKS..]', 'synth [STACKS..]' ], 'Synthesizes and prints the CloudFormation template for this stack', yargs => yargs
       .option('exclusively', { type: 'boolean', alias: 'e', desc: 'only deploy requested stacks, don\'t include dependencies' }))
     .command('bootstrap [ENVIRONMENTS..]', 'Deploys the CDK toolkit stack into an AWS environment', yargs => yargs
-      .option('toolkit-bucket-name', { type: 'string', alias: 'b', desc: 'The name of the CDK toolkit bucket', default: undefined })
+      .option('toolkit-bucket-name', { type: 'string', alias: ['b', 'bootstrap-bucket-name'], desc: 'The name of the CDK toolkit bucket', default: undefined })
```
Comment (eladb) [inline packages/aws-cdk/bin/cdk.ts]: Sorry for being annoying, can you make bootstrap-bucket-name the primary?

Comment (hoegertn) [reply packages/aws-cdk/bin/cdk.ts]: Sure, give me a second. I am fixing the multiple LambdaIntegrations PR currently

Comment (hoegertn) [reply packages/aws-cdk/bin/cdk.ts]: Done!

**Pull request merged**